fedimint-core/src/encoding/mod.rs

```diff
@@ -970,6 +970,17 @@ impl Decodable for Cow<'static, str> {
     }
 }
 
+// Simple decoder implementing `bitcoin_io::Read` for `std::io::Read`.
+// This is needed because `bitcoin::consensus::Decodable` requires a
+// `bitcoin_io::Read`.
+pub struct SimpleBitcoinRead<R: std::io::Read>(R);
+
+impl<R: std::io::Read> bitcoin_io::Read for SimpleBitcoinRead<R> {
+    fn read(&mut self, buf: &mut [u8]) -> bitcoin_io::Result<usize> {
+        self.0.read(buf).map_err(bitcoin_io::Error::from)
+    }
+}
```

Comment:
Note for reviewers: We're moving this to a more central place since it will be useful for implementing `Decodable` across the entire codebase later in the bitcoin v0.32 migration process